### PR TITLE
Hadi

### PR DESCRIPTION
**Pull request by hadign20**
Please merge these changes

### CODE REVIEW

File: CMG-C++/main.cpp [hunk 1]
```diff
@@ -9,9 +9,9 @@ void main()
...
 	vector<shlevel>  H;
 	int flag = 0;
-	int opts = 0;
+	int opts = 1;
 	matrix A;
-	A.n = 450;
+	A.n = 550;
 	cmg_precondition(pfun, H, flag,A,  opts);
-
+	system("Pause");
```
Comment (hessam94) [inline CMG-C++/main.cpp:16]: please remove this line

**Review by hessam94 — CHANGES_REQUESTED** [review]

File: CMG-C++/main.cpp (hunk 1 above)
Comment (hadign20) [reply CMG-C++/main.cpp:16]: No

Comment (hadign20) [reply CMG-C++/main.cpp:16]: line

**Review by hessam94 — APPROVED** [review]

**Comment by hessam94** [comment]
new function such as validate has been declared from hadi branch.

**Pull request merged**